File: schemaorg/authorizeaction.go
package schemaorg

import "encoding/json"

// AuthorizeAction see : https://schema.org/AuthorizeAction
type AuthorizeAction struct {

typeContext

AllocateAction

// Recipient see : https://schema.org/recipient
// A sub property of participant. The participant who is at the receiving end of the action.
Recipient interface{} `json:"recipient"` // types : Audience ContactPoint Organization Person

}

func (v *AuthorizeAction) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "AuthorizeAction"

	return json.Marshal(v)
}

File: schemaorgphp/classes/authorize_action.php
<?php

class AuthorizeAction extends AllocateAction implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'AuthorizeAction';
	
	/**
	 * A sub property of participant. The participant who is at the receiving end of the action.
	 * see : https://schema.org/recipient
	 * @var \Audience|\Audience[]|\ContactPoint|\ContactPoint[]|\Organization|\Organization[]|\Person|\Person[]
	 */
	public var $recipient;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'AuthorizeAction'
		);
		
		$serialized = so_json_serialize( $this->recipient );
		if ( ! empty( $serialized ) ) {
			$out['recipient'] = $serialized;
		}
		
		return $out;
	}
}